src/engine.cc

```diff
@@ -0,0 +1,83 @@
+#include "engine.h"
+#include "gobject.h"
+
+#include <eos-knowledge-content.h>
+
+using namespace v8;
+
+struct CallbackData {
+  Persistent<Promise::Resolver> resolver;
+};
+
+static void EngineGetObjectCallback(GObject *source,
+                                    GAsyncResult *result,
+                                    gpointer user_data)
+{
+  Isolate *isolate = Isolate::GetCurrent ();
+  HandleScope scope(isolate);
+  CallbackData *data = (CallbackData *)user_data;
+
+  GError *error = NULL;
+  EkncContentObjectModel *model;
+
+  Local<Promise::Resolver> resolver = Local<Promise::Resolver>::New(isolate, data->resolver);
```

Comment:
Yeah, wasn't able to get this working. And most v8 async examples I can find seem to make a local from a persistent and call that.